core/unit_test/tstLinkedCellList.hpp

```diff
@@ -237,6 +236,437 @@ void testLinkedList()
+        Cabana::LinkedCellList<typename AoSoA_t::memory_space>
+            cell_list( pos, grid_delta, grid_min, grid_max );
+        Cabana::permute( cell_list, aosoa, begin, end );
+
+        // Checking the binning.
+        EXPECT_EQ( cell_list.totalBins(), nx*nx*nx );
+        EXPECT_EQ( cell_list.numBin(0), nx );
+        EXPECT_EQ( cell_list.numBin(1), nx );
+        EXPECT_EQ( cell_list.numBin(2), nx );
+
+        // The order should be reversed with the i index moving the slowest
+        // for those that are actually in the binning range. Do this pass
+        // first. We do this by looping through in the sorted order and check
+        // those that had original indices in the sorting range.
+        particle_id = 0;
+        for ( int i = 0; i < nx; ++i )
+        {
+            for ( int j = 0; j < nx; ++j )
+            {
+                for ( int k = 0; k < nx; ++k, ++particle_id )
+                {
+                    if ( begin <= particle_id && particle_id < end )
+                    {
+                        // Back calculate what we think the ijk indices of
+                        // the particle are based on k-fastest ordering.
+                        int grid_id = i * nx * nx + j * nx + k;
+                        int grid_i = grid_id / (nx*nx);
+                        int grid_j = (grid_id / nx) % nx;
+                        int grid_k = grid_id % nx;
+
+                        // Check the indices of the particle
+                        EXPECT_EQ( cell_id( particle_id, 0 ), grid_i );
+                        EXPECT_EQ( cell_id( particle_id, 1 ), grid_j );
+                        EXPECT_EQ( cell_id( particle_id, 2 ), grid_k );
+
+                        // Check that we binned the particle and got the right
+                        // offset.
+                        EXPECT_EQ( cell_list.binSize(i,j,k), 1 );
+                        EXPECT_EQ( cell_list.binOffset(i,j,k),
+                                   size_type(particle_id) );
+                    }
+                }
+            }
+        }
+
+        // For those that are outside the binned range IDs should be
+        // unchanged
+        particle_id = 0;
+        for ( int k = 0; k < nx; ++k )
+        {
+            for ( int j = 0; j < nx; ++j )
+            {
+                for ( int i = 0; i < nx; ++i, ++particle_id )
+                {
+                    if ( begin > particle_id || particle_id >= end )
+                    {
+                        EXPECT_EQ( cell_id( particle_id, 0 ), i );
+                        EXPECT_EQ( cell_id( particle_id, 1 ), j );
+                        EXPECT_EQ( cell_id( particle_id, 2 ), k );
+                        EXPECT_EQ( cell_list.binSize(i,j,k), 1 );
+                    }
+                }
+            }
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+void testLinkedListSlice()
+{
+    // Make an AoSoA with positions and ijk cell ids.
```

Comment:
Same here in terms of code reuse for initialization

Reply:
Right now code is duplicated (especially throughout tests for Sort); what do you think about adding back test fixtures where it makes sense?